Assets/Scripts/LetterBlock.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using UnityEngine;
+
+/// <summary>
+/// Choose letter block color, handle shake and break animation
+/// </summary>
+public class LetterBlock : MonoBehaviour
+{
+    [SerializeField] private GameObject letter;
+    [SerializeField] private GameObject particles;
+    [SerializeField] private float blockBreakDelay;
+    [SerializeField] [Range(0f, 0.1f)] private float delayBetweenShakes = 0f;
+    [SerializeField] [Range(0f, 2f)] private float distance = 0.1f;
+
+    private SpriteRenderer _renderer;
+    private Vector3 _startPos;
+    private float _timer;
+
+    private void Awake()
+    {
+        _startPos = transform.position;
+        _renderer = GetComponent<SpriteRenderer>();
+    }
+
+    public void Crack()
+    {
+        particles.SetActive(true);
+        StartCoroutine(Shake());
+        StartCoroutine(Break());
+    }
+    
+    private IEnumerator Shake()
+    {
+        _timer = 0f;
+
+        while (_timer < blockBreakDelay)
+        {
+            _timer += Time.deltaTime;
```

Comment:
nit: `_timer` could be a local variable rather than a field (e.g. `for (float timer = 0; timer < blockBreakDelay; timer += Time.deltaTime)`)